subsys/mgmt/mcumgr/smp.c

```diff
@@ -117,6 +117,10 @@ zephyr_smp_split_frag(struct net_buf **nb, void *arg, uint16_t mtu)
 		frag = src;
 	} else {
 		frag = zephyr_smp_alloc_rsp(src, arg);
+		if (!frag) {
+			zephyr_smp_free_buf(src, arg);
+			return NULL;
+		}
```

Comment:
I think that freeing `src` should be a responsibility of `zephyr_smp_split_frag()` caller. This is also what the example in `zephyr_smp_split_frag()` documentation says:
```
 * This function is expected to be called in a loop until the entire source
 * net_buf has been consumed.  For example:
 *
 *     struct net_buf *frag;
 *     struct net_buf *rsp;
 *     ...
 *     while (rsp != NULL) {
 *         frag = zephyr_smp_split_frag(&rsp, zst, get_mtu());
 *         if (frag == NULL) {
 *             net_buf_unref(nb);
 *             return SYS_ENOMEM;
 *         }
 *         send_packet(frag)
 *     }
```

Reply:
What should be fred here?

Reply:
In `zephyr_smp_split_frag()` nothing should be freed, as it just fails to split message. The logic to free the whole message, if it cannot be split, should be caller responsibility.

Reply:
Changed